File: src/bmsx_cpp/core/system.cpp
#include "system.h"

#include "machine/runtime/timing/constants.h"

namespace bmsx {
namespace {

const MachineManifest SYSTEM_MACHINE_MANIFEST = [] {
	MachineManifest manifest;
	manifest.namespaceName = "bmsx";
	manifest.viewportWidth = 256;
	manifest.viewportHeight = 212;
	manifest.ufpsScaled = DEFAULT_UFPS_SCALED;
	manifest.cpuHz = 1'000'000;
	manifest.imgDecBytesPerSec = 26'214'400;
	manifest.dmaBytesPerSecIso = 8'388'608;
	manifest.dmaBytesPerSecBulk = 26'214'400;
	manifest.vdpWorkUnitsPerSec = DEFAULT_VDP_WORK_UNITS_PER_SEC;
	manifest.geoWorkUnitsPerSec = DEFAULT_GEO_WORK_UNITS_PER_SEC;
	return manifest;
}();

} // namespace

const MachineManifest& defaultSystemMachineManifest() {
	return SYSTEM_MACHINE_MANIFEST;
}

const char* systemBootEntryPath() {
	return "bios/bootrom";
}

} // namespace bmsx

File: src/bmsx/core/system.ts
import { DEFAULT_GEO_WORK_UNITS_PER_SEC, type MachineManifest } from '../rompack/format';
import { DEFAULT_UFPS_SCALED } from '../machine/runtime/timing/constants';

export const SYSTEM_ROM_NAME = 'bmsx-bios';
export const SYSTEM_BOOT_ENTRY_PATH = 'bios/bootrom.lua';

export const SYSTEM_MACHINE_MANIFEST: MachineManifest = {
	render_size: {
		width: 256,
		height: 212,
	},
	namespace: 'bmsx',
	ufps: DEFAULT_UFPS_SCALED,
	specs: {
		cpu: {
			cpu_freq_hz: 1_000_000,
			imgdec_bytes_per_sec: 26_214_400,
		},
		dma: {
			dma_bytes_per_sec_iso: 8_388_608,
			dma_bytes_per_sec_bulk: 26_214_400,
		},
		vdp: {
			work_units_per_sec: 25_600,
		},
		geo: {
			work_units_per_sec: DEFAULT_GEO_WORK_UNITS_PER_SEC,
		},
	},
};
